float64 x
float64 y
float64 z
bool perform_o_pattern
---
bool success